smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/auth/AuthUtils.java

```diff
@@ -149,4 +150,23 @@ public static Map<String, HttpAuthSchemeParameter> collectHttpAuthSchemeParamete
         }
         return httpAuthSchemeParameters;
     }
+
+    public static boolean areHttpAuthSchemesEqual(
+        Map<ShapeId, Trait> httpAuthSchemes1,
+        Map<ShapeId, Trait> httpAuthSchemes2
+    ) {
+        if (httpAuthSchemes1.size() != httpAuthSchemes2.size()) {
+            return false;
+        }
+        var iter1 = httpAuthSchemes1.entrySet().iterator();
+        var iter2 = httpAuthSchemes2.entrySet().iterator();
+        while (iter1.hasNext() && iter2.hasNext()) {
+            var entry1 = iter1.next();
+            var entry2 = iter2.next();
+            if (!entry1.equals(entry2)) {
+                return false;
```

Comment:
if order is important the method interfaces involved or leading up to the point of comparison should indicate a return type of the actual ordered map implementation instead of `Map`. 

For example

```java
Map<ShapeId, Trait> getEffectiveAuthSchemes
```

should become

```java
LinkedHashMap<ShapeId, Trait> getEffectiveAuthSchemes
```

Reply:
The `Map`s returned in Smithy are not more specifically typed so we can't assume a more specific type.

Typically, `Map` equality in Java only checks if all of the items in 1 map is in the other and vice versa, and not order, so this explicit iterator check (albeit kind of a black box) is necessary.